### PR TITLE
Add Docker container

### PR DESCRIPTION
**Pull request by AaronDewes**
These Docker containers are useful when using this with LND and will be required for integration into Umbrel and Citadel.

They are automatically build in GitHub Actions and pushed to the GitHub container registry.

### CODE REVIEW

**Comment by sputn1ck** [comment]
Thanks for the PR.

Does it make sense to have a docker container built every commit or should we only have a container built on tagged releases?  @nepet

**Comment by nepet** [comment]
> Thanks for the PR.
> 
> Does it make sense to have a docker container built every commit or should we only have a container built on tagged releases? @nepet

I guess it makes sense to have fixed versions on tagged releases as well as a `master` or `develop` version where people could test the container on their Umbrel or Citadel as well as for the Umbrel and Citadel developers. Best would be for the Umbrel and Citadel UIs to only allow the `develop` versions after def. call to action from the user.

I'll try to review this pr until end of the week, if desired.

**Comment by nepet** [comment]
On the other hand, developers could build a `master` container from source, this way we would make it harder to misuse potentially untested versions.

**Comment by nepet** [comment]
Nice addition that you now only push them on tag releases and when integtration tests pass!
`LGTM`

**Comment by sputn1ck** [comment]
There is still some docker code in the `ci.yaml` file. Is that intended?

I think we don't really need to run integration tests twice on the same commit, which would happen if run integration tests on push and on tagging. As tagging is manual we would only tag if the integration tests pass right?

**Comment by nepet** [comment]
> @nepet I think we don't really need to run integration tests twice on the same commit, which would happen if run integration tests on push and on tagging. As tagging is manual we would only tag if the integration tests pass right?

In theory, yes! Good point.

**Comment by sputn1ck** [comment]
But as open source repositories are free, there is no reason not to run on tags. So that's fine with me.

**Comment by nepet** [comment]
Might also help on unintentional push to a tag from someone with write access to the repo ^^

**Comment by AaronDewes** [comment]
> @AaronDewes There is still some docker code in the `ci.yaml` file. Is that intended?

No, I remove it in a few minutes.

File: .github/workflows/tag.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+      - name: Run go tests
+        run: make test
+
+  integration:
+    runs-on: ubuntu-latest
+    needs: [buildandtest]
+    strategy:
+      max-parallel: 4
+      matrix:
+        test-vector: [bitcoin-cln, bitcoin-lnd, liquid-cln, liquid-lnd]
```
[the remark this answers is not in the document]
Comment (sputn1ck) [reply .github/workflows/tag.yml:38]: good point, if we add some cln related stuff in the release we can readd it

**Pull request merged**